python/ray/serve/_private/long_poll.py

```diff
@@ -247,24 +249,29 @@ async def listen_for_change(
         immediately if the snapshot_ids are outdated, otherwise it will block
         until there's an update.
         """
-        watched_keys = keys_to_snapshot_ids.keys()
-        existent_keys = set(watched_keys).intersection(set(self.snapshot_ids.keys()))
-
         # If there are any keys with outdated snapshot ids,
         # return their updated values immediately.
-        updated_objects = {
-            key: UpdatedObject(self.object_snapshots[key], self.snapshot_ids[key])
-            for key in existent_keys
-            if self.snapshot_ids[key] != keys_to_snapshot_ids[key]
-        }
+        updated_objects = {}
+        for key, snapshot_id in keys_to_snapshot_ids.items():
+            try:
+                existing_id = self.snapshot_ids[key]
+            except KeyError:
+                # The caller may ask for keys that we don't know about (yet),
+                # just ignore them.
```

Comment:
When does this happen?

Reply:
I think this can happen when e.g. a `DeploymentHandle` is created for a deployment that may not exist yet - normally that won't happen if you're going through `serve.get_app_handle`, but it can happen if you're making `DeploymentHandle`s manually or bypassing the checks (which is something we do do in our use case :) )

Reply:
Got it, could you leave a comment mentioning that?

Reply:
Can do!